### PR TITLE
Make it easier to run the tool locally

### PR DESCRIPTION
**Pull request by RikkiGibson**
This PR adds a `/local` flag to run the tool locally. It prompts you with a nice popup login window. This way you don't have to type your corporate username and password in the command line/save it in VS debug options/other bad stuff like that.

TODO: make pushing the PR branch locally work. I suspect the client credential uses a token but libgit2sharp doesn't expose any way to provide a custom header with the token for a git push. So we would have to either:

1. send a patch to libgit2sharp to add the option (probably not that hard, since they did do the native bridging in libgit2/libgit2sharp#1233)
2. just `Process.Start("git", ...).WaitForExit()`

/cc @JoeRobich @dibarbet @jmarolf

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs [hunk 1]
```diff
@@ -407,14 +414,13 @@ private static EnvironmentStatus WaitForReleaseEnvironmentCompletion(string proj
...
-                    Task.Delay(5 * 1000, cancellationToken).Wait();
+                    Task.Delay(5000, cancellationToken).Wait();
 
                     environmentStatus = GetReleaseEnvironment(projectName, release.Id, environmentId, releaseClient, cancellationToken).Status;
 
-                    if(!(environmentStatus.Equals(EnvironmentStatus.InProgress)
-                          || environmentStatus.Equals(EnvironmentStatus.Queued)
-                          || environmentStatus.Equals(EnvironmentStatus.NotStarted)
-                          || environmentStatus.Equals(EnvironmentStatus.Scheduled)))
+                    if (environmentStatus != EnvironmentStatus.InProgress
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: this change keeps the tool from spinning for 10 minutes waiting for a release that isn't actually happening.

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs [hunk 2]
```diff
@@ -296,6 +296,10 @@ public static partial class RoslynInsertionTool
                                 Console.WriteLine("Could not create diff links.");
                             }
 
+                            Console.WriteLine("---PR description---");
+                            Console.WriteLine(prDescription);
```
Comment (RikkiGibson) [inline src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs]: Eventually I'd like to have a dry-run flag for local testing so we can have decent experience finally getting to the bottom of #502

File: src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs (hunk 1 above)
Comment (dibarbet) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: Do we need any of this release stuff at all?  What is the release for?

Comment (RikkiGibson) [reply src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.VisualStudioTeamServices.cs]: The comments make it seem like it is supposed to do symbol publishing for the build we're inserting. But it's not clear to me if that is still something that RIT users need to be worried about, or whether it's something that just happens as part of the signed build pipeline. Can you shed any light @brettfo?

**Comment by RikkiGibson** [comment]
closing in lieu of #570

**Pull request closed without merging**